File: Programación Orientada a Objetos (PDOO)/Napakalaki(Java)/src/napakalaki/NumericBadConsequence.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package napakalaki;

import java.util.ArrayList;

/**
 *
 * @author joseperoman
 */
public class NumericBadConsequence extends BadConsequence{
    private int nVisibleTreasures;
    private int nHiddenTreasures;
    
    public NumericBadConsequence(String txt, int lvl,int nv, int nh){
    super(txt,lvl);
    nVisibleTreasures=nv;
    nHiddenTreasures=nh;
}
    
    public int getnVisibleTreasures(){
        return nVisibleTreasures;
    }
    public int getnHiddenTreasures(){
        return nHiddenTreasures;
    }
    
    @Override
    public boolean isEmpty(){
        boolean empty=false;
        if(nVisibleTreasures == 0 && nHiddenTreasures==0){
            empty=true;
        }
        return empty;
    }
    @Override
    public void substractVisibleTreasure(Treasure t){
        if(nVisibleTreasures>0){
            nVisibleTreasures-=1;
        }
    }
    @Override
    public void substractHiddenTreasure(Treasure t){
        if(nHiddenTreasures>0){
            nHiddenTreasures-=1;
        }
    }
    @Override
    public NumericBadConsequence adjustToFitTreasureList(ArrayList<Treasure> v, ArrayList<Treasure> h){
        int nvisible= 0, nhidden=0;
        
        NumericBadConsequence nbc = new NumericBadConsequence("",0,0,0);
        
        if(v.size() <= this.nVisibleTreasures){
            nvisible=v.size();
        }
        else nvisible=this.nVisibleTreasures;
            
        
        if(h.size() <= nHiddenTreasures){
            nhidden=h.size();
        }
        else nhidden=this.nHiddenTreasures;
        
        nbc= new NumericBadConsequence(this.getText(),this.getLevels(),nvisible,nhidden);
        
        return nbc;
    }
    
    
    public String toString(){
        return /*" \n\tTexto = " + text + */" \n\tLevels = " + Integer.toString(levels) + " \n\tnVisibleTreasures =  " + Integer.toString(nVisibleTreasures) + " \n\tnHiddenTreasures = " + Integer.toString(nHiddenTreasures);
    }
}

File: Programación Orientada a Objetos (PDOO)/Napakalaki(Ruby)/pkg/pruebaNapakalaki-0.0.1/lib/numeric_bad_consequence.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "bad_consequence.rb"
module NapakalakiGame
class NumericBadConsequence < BadConsequence

  attr_accessor :nVisibleTreasures, :nHiddenTreasures
  def initialize(txt,lvls,nV,nH)
    super(txt,lvls)
    #@text=txt
    #@levels=lvls
    @nVisibleTreasures=nV
    @nHiddenTreasures=nH
  end
 
  def isEmpty()
    empty = false
    if @nVisibleTreasures==0 && @nHiddenTreasures==0 then
        empty=true
    end
      return empty
  end
  
  def substractVisibleTreasures(t)
    if (@nVisibleTreasures > 0) then
       @nVisibleTreasures-=1;
    end 
  end
  
  def substractHiddenTreasures(t)
    if(@nHiddenTreasures > 0)
      @nHiddenTreasures-=1; 
    end
  end
  
  def setnVisibleTreasures(nV)
    @nVisibleTreasures = nV
  end
  
  def setnHiddenTreasures(nH)
    @nHiddenTreasures = nH
  end
  
  def adjustToFitTreasureList(v,h)
    
    nvisible=0
    nhidden=0
  
    if @nVisibleTreasures !=0 || @nHiddenTreasures !=0 then
              
      if(v.size <= @nVisibleTreasures) then
        nvisible = v.size();                      
      else
        nvisible = @nVisibleTreasures;
      end

      if (h.size() <= @nHiddenTreasures) then
        nhidden= h.size()
      else
        nhidden = @nHiddenTreasures
      end
        
      bd = NumericBadConsequence.new(@text, 0, nvisible, nhidden)
    end
    
    bd
  end
  
   def to_s
    "#{@text}\n Niveles perdidos: #{@levels}\n Tesoros visibles: #{@nVisibleTreasures}\n Tesoros ocultos: #{@nHiddenTreasures}"
  end
  
end
end